File: targets/python-web/vulns/vuln06_path.py
from flask import Blueprint, request
bp = Blueprint("vuln06", __name__)

@bp.get("/vuln06/download")
def download():
    name = request.args.get("name", "")
    # VULN:VULN-06:CWE-22:taint path traversal — user input concatenated into a filesystem path
    with open("/app/data/" + name, "rb") as f:
        return f.read()

File: targets/jsts-web/src/vulns/vuln06_path.ts
import { Router } from "express";
import * as fs from "fs";
export const router = Router();
router.get("/vuln06/download", (req, res) => {
  const name = String(req.query.name || "");
  // VULN:VULN-06:CWE-22:taint path traversal — user input concatenated into a filesystem path
  res.send(fs.readFileSync("/app/data/" + name));
});
